clarity/tests/collateral-rebalancing-pool_test.ts

```diff
@@ -0,0 +1,294 @@
+        call.result.expectOk().expectUint(5474288282941)
+        
+        // let's check what is the weight to wbtc (token)
+        call = await CRPTest.getWeightY(wbtcAddress, usdaAddress, expiry, 50000 * ONE_8, bs_vol);
+        call.result.expectOk().expectUint(33465993);          
+
+        // wbtc (token) falls by 20% vs usda (collateral)
+        oracleresult = Oracle.updatePrice(deployer,"WBTC","nothing",wbtcPrice * 0.8);
+        oracleresult.expectOk()
+
+        // move forward by one day
+        chain.mineEmptyBlockUntil(144);
+
+        // with wbtc (token) falling, weight to wbtc (token) should decrease
+        call = await CRPTest.getWeightY(wbtcAddress, usdaAddress, expiry, 50000 * ONE_8, bs_vol);
+        call.result.expectOk().expectUint(25427001);              
+
+        // arbtrageur selling 100 usda for wbtc
+        result = CRPTest.swapXForY(deployer, wbtcAddress, usdaAddress, expiry, 100 * ONE_8);
+        position = result.expectOk().expectTuple();
+        position['dx'].expectUint(100 * ONE_8);
+        position['dy'].expectUint(201315);         
+
+        // swap triggers weight change
+        call = await CRPTest.getPoolDetails(wbtcAddress, usdaAddress, expiry);
+        position = call.result.expectOk().expectTuple();
+        position['weight-y'].expectUint(25427001);          
+        position['weight-x'].expectUint(ONE_8 - 25427001);              
+        position['yield-supply'].expectUint(88881023);
+        position['key-supply'].expectUint(88881023);   
+        
+        // simulate to half way to expiry
+        chain.mineEmptyBlockUntil((expiry / ONE_8) / 2)    
+
+        // wbtc rises then by 50%
+        oracleresult = Oracle.updatePrice(deployer,"WBTC","nothing",wbtcPrice * 0.8 * 1.5);
+        oracleresult.expectOk();
+
+        // the rise shifts allocation to wbtc (token)
+        call = await CRPTest.getWeightY(wbtcAddress, usdaAddress, expiry, 50000 * ONE_8, bs_vol);
+        call.result.expectOk().expectUint(73683600);         
+        
+        // arbtrageur selling 100 usda for wbtc
+        result = CRPTest.swapXForY(deployer, wbtcAddress, usdaAddress, expiry, 100 * ONE_8);
+        position = result.expectOk().expectTuple();
+        position['dx'].expectUint(100 * ONE_8);
+        position['dy'].expectUint(294164);       
+
+        // swap triggers weight change - but not too much given moving average
+        call = await CRPTest.getPoolDetails(wbtcAddress, usdaAddress, expiry);
+        position = call.result.expectOk().expectTuple();
+        position['weight-y'].expectUint(73683600);            
+        position['weight-x'].expectUint(ONE_8 - 73683600);            
+        position['yield-supply'].expectUint(88881023);
+        position['key-supply'].expectUint(88881023);  
+        
+        // simulate to 3 / 4 to expiry
+        chain.mineEmptyBlockUntil((expiry / ONE_8) * 3 / 4)    
+
```

Comment:
I think from this part we can make to other independent testcase?? 
Do you think testing in a single flow is better?